src/js/components/Tab/Tab.js

```diff
@@ -13,118 +13,107 @@ import { normalizeColor } from '../../utils';
 
 import { StyledTab } from './StyledTab';
 
-class Tab extends Component {
-  static getDerivedStateFromProps(nextProps, prevState) {
-    const { active } = nextProps;
-    const { over } = prevState;
-    if (active && over) {
-      return { over: undefined };
-    }
-    return null;
-  }
-
-  state = {};
-
-  onMouseOver = event => {
-    const { onMouseOver } = this.props;
-    this.setState({ over: true });
+const Tab = props => {
+  const {
+    active,
+    forwardRef,
+    plain,
+    title,
+    onMouseOver,
+    onMouseOut,
+    theme,
+    ...rest
+  } = props;
+
+  delete rest.onActivate;
+
+  let normalizedTitle = title;
+  const tabStyles = {};
+
+  const [state, setState] = useState({
+    active: props && active ? active : false,
+    over: undefined,
+  });
+
+  const onMouseOverTab = event => {
+    setState({ over: true });
     if (onMouseOver) {
       onMouseOver(event);
     }
   };
 
-  onMouseOut = event => {
-    const { onMouseOut } = this.props;
-    this.setState({ over: undefined });
+  const onMouseOutTab = event => {
+    setState({ over: undefined });
     if (onMouseOut) {
       onMouseOut(event);
     }
   };
 
-  onClickTab = event => {
-    const { onActivate } = this.props;
+  const onClickTab = event => {
+    const { onActivate } = props;
```

Comment:
don't  have time to test this right now but i am not sure why we do this. why don't we destructure onActivate always and just call the onActivate if it exists? do we forward this later to something?